### PR TITLE
Add cosmos tf and diagram

### PR DESCRIPTION
**Pull request by soferreira**
Cosmos tf files added.
Created one resource groups resource group per region as cosmos pe need to be in the same region will be created in the region of its vnet.
Added architecture diagram to readme.

### CODE REVIEW

File: comsos.tfvars [hunk 1]
```diff
@@ -0,0 +1,70 @@
+cosmos_account_name = "samplecosmosacct202"
```
Comment (zioproto) [inline comsos.tfvars]: change file name comsos to cosmos

File: cosmos.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Azure provider version
```
Comment (zioproto) [inline cosmos.tf]: You should merge this block with what is already available in provider.tf

**Comment by soferreira** [comment]
@zioproto new changes committed, please review again

**Comment by zioproto** [comment]
You added more variables in a different file than `.tfvars`. 
Can you add in the README that to run the project you need:

```
 terraform apply -var-file=.tfvars -var-file=cosmos.tfvars
```

Or just consolidate all variables in a single file. Thanks

**Pull request merged**